kernel/src/platform/mpu.rs

```diff
@@ -49,6 +49,20 @@ impl Region {
     }
 }
 
+/// Null type for the default type of the `MpuConfig` type in an implementation
+/// of the `MPU` trait. We need this to workaround a bug in the Rust compiler.
+///
+/// If https://github.com/rust-lang/rust/issues/65774 is resolved then we should
```

Comment:
My assumption is that the resolution of rust-lang/rust#65774 is not going to include making `()` implement `Display`.

That is, if that bug is fixed and you revert this commit, then what you will get is a proper compiler error at the point where you are doing:

```rust
type MpuConfig: Default + Display = ();
```

My suggestion is that you just keep this `MpuConfigDefault` type. (That, or use something other than `fmt::Display` as your bound on the `MpuConfig` associated type.)